sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -536,6 +536,8 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
 
     case a: AlterTableCommand if a.table.resolved =>
       val table = a.table.asInstanceOf[ResolvedTable]
+      ResolveTableConstraints.validateCatalogForTableChange(
```

Comment:
Note: there will be new tests for this one when the `alter table with constraint` is supported end-to-end.